Gemfile

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 source 'https://rubygems.org'
+ruby ENV['CUSTOM_RUBY_VERSION'] || '2.5.1'
```

Comment:
Bu düzenleme patchlevel'ları yönetiyor mu?  Örneğin şu an benim makinede durum:

```sh
$ ruby -v
ruby 2.5.1p57
```

omu imajlarında da aynı sürüm ve patchlevel (`p57`) kurulu.  Bunu deneyebildin mi?  Patchlevel'dan rahatsız olursa bir versiyon operatörü kullanmamız gerekecek, o yüzden soruyorum.

Reply:
Bu düzenleme patchlevel'ları yönetemiyor. Gemfile'da ruby versiyonu aşağıdaki formatta [veriliyormuş](https://bundler.io/v1.12/gemfile_ruby.html):

```ruby
ruby 'RUBY_VERSION', :engine => 'ENGINE', :engine_version => 'ENGINE_VERSION',
  :patchlevel => 'RUBY_PATCHLEVEL'
```

Reply:
`:patchlevel` seçeneği verilirse ortamdan gelen versiyonla karışacak bu sefer.  Aşağıdaki düzenleme çalışacaktır sanırım: `2.5.1` sürümüne eşit veya büyük

```ruby
ruby ENV['CUSTOM_RUBY_VERSION'] || '~> 2.5.1'
```

Reply:
> omu imajlarında da aynı sürüm ve patchlevel (p57) kurulu. Bunu deneyebildin mi?

buildpack de p57 kuruyor. Heroku tarafından(yani heroku buildpack'ler tarafından) desteklenen versiyonlar: https://devcenter.heroku.com/articles/ruby-support#supported-runtimes

Reply:
Yani yönetiyor (hata vermiyor)?  Yukarıda yönetemiyor dediğinden yazdım. :-)  Desteklenen sürümleri biliyorum, tek emin olamadığım Gemfile'daki `ruby '2.5.1'` versiyon bildirimi `ruby 2.5.1p57` sürümlü klasik dev ortamında veya Dokku'da şikayetsiz çalışıyor mu?

Reply:
Vagrant'ta deneyerek, sorumun cevabını aldım.  `ruby '2.5.1'` versiyon bildirimi patchlevel'ları da kapsıyor.  Örneğin bundle `2.5.1p57`'yi `2.5.1.57` gibi görüyor ve dördüncü seviyeyi (patch level) dikkate almıyor.  Geçmişte bir ara bundle'dan beklemediğim bir hata almıştım o yüzden kuşkulandım.

Yönetme konusunda derdimi tam anlatamadım, kusura bakma. Bu durumda sorun kalmadı.  PR'ı alabiliriz.

Reply:
> Yani yönetiyor (hata vermiyor)? Yukarıda yönetemiyor dediğinden yazdım. :-)
Aslında şu yazımı yönetemiyor demek istemiştim, ruby '2.5.1-p57' ama açık bir şekilde belirtmemişim :(

ruby '2.5.1' versiyon bildirimi, hem dev ortamında hem de dokku'da sorunsuz çalışıyor, her iki ortamda da p57 geliyor.